client/extensions/woocommerce/state/data-layer/products/index.js

```diff
@@ -75,3 +83,22 @@ export function handleProductRequest( { dispatch }, action ) {
 	const updatedSuccessAction = updatedAction( siteId, action, successAction );
 	dispatch( get( siteId, 'products/' + productId, updatedSuccessAction, failureAction ) );
 }
+
+export function handleProductsRequestSuccess( store, action, next, response ) {
+	const { body, status } = response.data;
+
+	// We needed to envelope the reponse to get the number of products and pages. see handleProductsRequest
+	if ( status !== 200 ) {
+		action.meta.dataLayer.error = { code: ( body.code || status ), message: body.message };
```

Comment:
I'll set this back to `In Progress` for now, pending our chat. If I remove the last commit, this uses the `*_RECEIVE` pattern, but I'll hold off doing anything until we figure out what direction we want to go in.